lib/Service/WorkspaceService.php

```diff
@@ -30,7 +30,10 @@ public function __construct(IL10N $l10n) {
 	public function getFile(Folder $folder): ?File {
 		foreach ($this->getSupportedFilenames() as $filename) {
 			try {
-				$file = $folder->get($filename);
+				$exists = $folder->getStorage()->getCache('')->get($filename);
```

Comment:
why the separate check here? if the file doesn't exist `get` throws a `NotFoundException` which is catched below.

Thus just duplicated the number of times we're querying the filecache

Reply:
The reason for this issue was slow performance with federated shares or external webdav storages where the additional check on the cache turned out faster then getting the file from the remote, but agree that we probably should not do that extra.